sound/soc/sof/ops.h

```diff
@@ -387,10 +387,47 @@ static inline int snd_sof_dma_trace_release(struct snd_sof_dev *sdev)
 
 static inline int snd_sof_dma_trace_trigger(struct snd_sof_dev *sdev, int cmd)
 {
-	if (sof_ops(sdev)->trace_trigger)
-		return sof_ops(sdev)->trace_trigger(sdev, cmd);
+	int ret = 0;
 
-	return 0;
+	if (!sdev->dtrace_is_supported)
+		return 0;
+
+	/* FIXME: return 0 or return -EINVAL ? */
```

Comment:
trace_trigger is an optional op, so return 0 please and remove FIXME and can you please combine the 2 checks for dtrace_is_supported and trace_trigger in 1

Reply:
ACK